src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -585,25 +601,25 @@ private void processLogsAndUpload() throws InterruptedException {
                             }
                         }
                     }
+                    // If there are no log files for the component, then return.
+                    if (allFiles.isEmpty()) {
+                        continue;
+                    }
                     // Sort the files by the last modified time. Then try to proceed oldest file first to avoid data
                     // loss caused by auto deletion
                     allFiles.sort(Comparator.comparingLong(LogFile::lastModified));
-                    // If there are no rotated log files for the component, then return.
-                    if (allFiles.size() - 1 <= 0) {
-                        continue;
-                    }
-                    // Don't consider the active log file.
-                    allFiles = allFiles.subList(0, allFiles.size() - 1);
+                    String activeFileName = allFiles.get(allFiles.size() - 1).getName();
```

Comment:
I'd rather call this again that storing the activeFileName. I generally want to try to avoid relying as much as possible on file names given we can't be sure how the customer is changing or doing with the names. Instead of storing this on the `ComponentLogFileInformation` has some construct like `LogDirectory` that you can ask it for the active file rather than relying on names to determine if something is active.